packages/anoncreds/src/utils/w3cAnonCredsUtils.ts

```diff
@@ -0,0 +1,201 @@
+  const anoncredsMetadata = record.metadata.get<W3cAnoncredsCredentialMetadata>(W3cAnonCredsCredentialMetadataKey)
+  if (!anoncredsMetadata) return undefined
+
+  const tags = record.getTags()
+  if (
+    !tags.anonCredsCredentialId ||
+    !tags.anonCredsLinkSecretId ||
+    !tags.anonCredsMethodName ||
+    !tags.anonCredsSchemaId ||
+    !tags.anonCredsSchemaName ||
+    !tags.anonCredsSchemaVersion ||
+    !tags.anonCredsSchemaIssuerId ||
+    !tags.anonCredsCredentialDefinitionId
+  ) {
+    return undefined
+  }
+
+  return Object.fromEntries(
+    Object.entries(tags).filter(([key]) => key.startsWith('anonCreds'))
+  ) as AnonCredsCredentialTags
+}
+
+export function getStoreCredentialOptions(
+  options: StoreCredentialOptions,
+  indyNamespace?: string
+): StoreCredentialOptions {
+  const {
+    credentialRequestMetadata,
+    credentialDefinitionId,
+    schema,
+    credential,
+    credentialDefinition,
+    revocationRegistry,
+  } = options
+
+  const storeCredentialOptions = {
+    credentialId: utils.uuid(),
+    credentialRequestMetadata,
+    credential,
+    credentialDefinitionId: isUnqualifiedCredentialDefinitionId(credentialDefinitionId)
+      ? getQualifiedDidIndyDid(credentialDefinitionId, indyNamespace as string)
+      : credentialDefinitionId,
+    credentialDefinition: isUnqualifiedDidIndyCredentialDefinition(credentialDefinition)
+      ? getQualifiedDidIndyCredentialDefinition(credentialDefinition, indyNamespace as string)
+      : credentialDefinition,
+    schema: isUnqualifiedDidIndySchema(schema) ? getQualifiedDidIndySchema(schema, indyNamespace as string) : schema,
+    revocationRegistry: revocationRegistry?.definition
+      ? {
+          definition: isUnqualifiedDidIndyRevocationRegistryDefinition(revocationRegistry.definition)
+            ? getQualifiedDidIndyRevocationRegistryDefinition(revocationRegistry.definition, indyNamespace as string)
+            : revocationRegistry.definition,
+          id: isUnqualifiedRevocationRegistryId(revocationRegistry.id)
+            ? getQualifiedDidIndyDid(revocationRegistry.id, indyNamespace as string)
+            : revocationRegistry.id,
+        }
+      : undefined,
+  }
+
+  return storeCredentialOptions
+}
+
+export function getW3cRecordAnonCredsTags(options: {
+  w3cCredentialRecord: W3cCredentialRecord
+  schemaId: string
+  schema: Omit<AnonCredsSchema, 'attrNames'>
+  credentialDefinitionId: string
+  revocationRegistryId?: string
+  credentialRevocationId?: string
+  linkSecretId: string
+  methodName: string
+}) {
+  const {
+    w3cCredentialRecord,
+    schema,
+    schemaId,
+    credentialDefinitionId,
+    revocationRegistryId,
+    credentialRevocationId,
+    linkSecretId,
+    methodName,
+  } = options
+
+  const issuerId = w3cCredentialRecord.credential.issuerId
+
+  const anonCredsCredentialRecordTags: AnonCredsCredentialTags = {
+    anonCredsCredentialId: w3cCredentialRecord.id,
```

Comment:
Same here. If it's the same as the id of the record, there's no need to add the tag